podcasts/EpisodesDataManager.swift

```diff
@@ -3,6 +3,30 @@ import PocketCastsServer
 import DifferenceKit
 
 class EpisodesDataManager {
+    // MARK: - Playlist episodes
+
+    /// Return the list of episodes for a given playlist
+    func episodes(for playlist: AutoplayHelper.Playlist) -> [BaseEpisode] {
+        switch playlist {
+        case .podcast(uuid: let uuid):
+            if let podcast = DataManager.sharedManager.findPodcast(uuid: uuid, includeUnsubscribed: true) {
+                return Array(episodes(for: podcast).compactMap { $0.elements as? [ListEpisode] }.joined()).map { $0.episode }
+            }
+        case .filter(uuid: let uuid):
+            if let filter = DataManager.sharedManager.findFilter(uuid: uuid) {
+                return episodes(for: filter).compactMap { $0.episode }
+            }
+        case .downloads:
+            return Array(downloadedEpisodes().compactMap { $0.elements }.joined()).map { $0.episode }
```

Comment:
We can skip the joining/Array casting step by using `flatMap`:

```suggestion
            return downloadedEpisodes().flatMap { $0.elements.map { $0.episode } }
```